actors/builtin/market/market_actor.go

```diff
@@ -636,18 +633,15 @@ func (a Actor) CronTick(rt Runtime, _ *abi.EmptyValue) *abi.EmptyValue {
 	return nil
 }
 
-func genRandNextEpoch(currEpoch abi.ChainEpoch, deal *DealProposal, rbF func(crypto.DomainSeparationTag, abi.ChainEpoch, []byte) abi.Randomness) (abi.ChainEpoch, error) {
-	buf := bytes.Buffer{}
-	if err := deal.MarshalCBOR(&buf); err != nil {
-		return epochUndefined, xerrors.Errorf("failed to marshal proposal: %w", err)
+func genRandNextEpoch(currEpoch abi.ChainEpoch, deal *DealProposal, dealID abi.DealID) (abi.ChainEpoch, error) {
+	offset := abi.ChainEpoch(uint64(dealID) % uint64(DealUpdatesInterval))
+	q := builtin.NewQuantSpec(DealUpdatesInterval, 0)
+	prevDay := q.QuantizeDown(deal.StartEpoch)
+	if prevDay+offset >= deal.StartEpoch {
+		return prevDay + offset, nil
```

Comment:
Shouldn't `QuantizeDown` only ever return `StartEpoch`?

And because `StartEpoch + offset` will always be `>= StartEpoch`, the second part of this method should never be hit.

Reply:
Nevermind - I didn't factor in the truncating division!